### DISCUSSION TITLE
Ghidra

### DISCUSSION THREAD

**Discussion by StanlsSlav** [catppuccin/catppuccin#2080] [Port Requests]
### Will you make the port?

- [X] Yes, I'm willing to make the port

### How can the suggested port be themed?

This repo already does it [zackelia/ghidra-dark-theme](https://github.com/zackelia/ghidra-dark-theme), but it's themed for FlatLeaf.

### Any additional comments?

Created a base/preview [repo](https://github.com/stanlsslav/ghidra) which has only the background with some built-in colors (such as red, yellow, pink ..) changed, but I'm yet to figure out how to properly theme Ghidra and do it via Catppuccin.

### Submission Guidelines

- [x] I have read and followed Catppuccin's submission guidelines.

**Comment by FabyoSK** [comment]
I'm interested on this theme, on what can help?

**Reply by StanlsSlav** [reply]
Hello! So far I've changed what's more straightforward, some of the [basic colors](https://github.com/StanlsSlav/ghidra/blob/main/CatppuccinMocha.theme#L537-L593) and searched for [exact matches](https://github.com/StanlsSlav/ghidra/blob/main/CatppuccinMocha.theme#L328-L338) on the language defaults table from [code editors](https://github.com/catppuccin/catppuccin/blob/main/docs/style-guide.md#code-editors) section of the style guide.

So far, this is the checklist:
1. Stylize the windows and their borders
2. Review for missing or incorrect usage of the palette
3. The used binary for the main preview, I'm not sure if it gives an extensive view of the theme 
4. Convert individual previews to webp format

**Reply by FabyoSK** [reply]
I'll look forward to `Stylize the windows and their borders` and create a draft PR.
Once I start it I'll let you know here

**Comment by FabyoSK** [comment]
@StanlsSlav 

I create a base PR improving just the mocha theme, you can check it [here](https://github.com/StanlsSlav/ghidra/pull/1)

A preview:
<img width="1280" alt="Screen Shot 2023-09-09 at 3 58 32 PM" src="https://github.com/catppuccin/catppuccin/assets/64048314/adfeab15-c4b6-45be-9311-d3cb0e96833e">

**Comment by FabyoSK** [comment]
Please provide feedback

**Reply by StanlsSlav** [reply]
A lot better! I'll update the other themes accordingly to your changes.

**Reply by FabyoSK** [reply]
Alright, I'll submit another PR today fixing the text and border color

**Reply by StanlsSlav** [reply]
Latte's a work on its own. How did you figure out which colors affect which parts of the UI? Colors such as `[color]laf.palette.color.73`
![image](https://github.com/catppuccin/catppuccin/assets/65411828/d41e1fbc-1285-43ee-b013-6bf0d82ded21)

**Reply by FabyoSK** [reply]
I used the raycast color picker and then search the color in the list

**Reply by FabyoSK** [reply]
@StanlsSlav Can you check my PR [here](https://github.com/StanlsSlav/ghidra/pull/2/files)?

**Reply by FabyoSK** [reply]
I'm focusing on Mocha because is the variant that I use, I don't have much experience using the other variants

**Reply by StanlsSlav** [reply]
Sure thing! Worry not, been using Mocha only as well.
I believe that the other dark themes should look good and be accessible only by changing the hex codes to the palette substitutes.

**Reply by FabyoSK** [reply]
Can you test the Mocha theme and provide feedback?
I've not seeing much to add style on it

**Reply by StanlsSlav** [reply]
Changing some colors from the palette changes the whole program. I believe it's mostly done, excluding listing and decompiler syntax.

So far I've seen missing only the graph views which use hardcoded hex colors.

**Reply by FabyoSK** [reply]
Yes, you are right,
I'll be busy in the next couple of days, in the weekend I'll help you with other themes

**Comment by FabyoSK** [comment]
And after I'll try to fix the syntax highlight with the proper color scheme

**Reply by StanlsSlav** [reply]
I'll get on with this part for Mocha, then translate for Frappé, Latte and Macchiato unless you'll get [some free time](https://github.com/catppuccin/catppuccin/discussions/2080#discussioncomment-6968432) and still want to continue with the theme.

**Comment by sgoudham** [comment]
Hey :wave:, I'm not sure how I missed this but recently [catppuccin/ghidra](https://github.com/catppuccin/ghidra) was transferred to the organisation a while ago.

Apologies, I didn't realise there was already an existing discussion with a draft of the port available. I'll be closing this as resolved since the repository exists but you may want upstream some of the contributions from your repository into the one underneath the organisation!
